api/handlers.go

```diff
@@ -359,6 +382,12 @@ func getVolumes(c *context, w http.ResponseWriter, r *http.Request) {
 			continue
 		}
 
+		if filters.Include("driver") {
```

Comment:
Instead of filtering in swarm, wouldn't it be possible to just have the engines do this filtering? The engine API already provides these filters, so should be able to return just what we're looking for?

(I'm not to familiar with the Swarm code-base, but always assumed that's what it did, but I now see all engine's functionality is replicated here)

Reply:
The filtering is done on locally cached volumes, making calls to all engines in the cluster to filter out accordingly defeats the swarm classic design. Note, this is not specific to Volumes, same applies to Images, Containers, Networks etc...

Reply:
Swarm classic is designed to always read from the cache (and do its own filtering) on read-only calls. For performance reasons, we cannot do engine calls during read-only operations.